File: hash-tables/javascript/check-voter.js
const VOTED = {};

function checkVoter(name) {
  if (VOTED[name]) {
    console.log(`${name}: kick them out!`);
  } else {
    VOTED[name] = true;
    console.log(`${name}: let them vote!`);
  }
}

checkVoter("tom");    // let them vote!
checkVoter("mike");   // let them vote!
checkVoter("mike");   // kick them out!

File: hash-tables/python/check-voter.py
# !/usr/bin/env python
# -*- coding: utf-8 -*-

voted = {}

def check_voter(name):
  if voted.get(name):
    print "{%s}: kick them out!" % name
  else:
    voted[name] = True
    print "{%s}: let them vote!" % name

check_voter("tom")    # let them vote!
check_voter("mike")   # let them vote!
check_voter("mike")   # kick them out!!

